engine/src/main/java/org/camunda/bpm/engine/impl/persistence/entity/JobManager.java

```diff
@@ -103,12 +103,7 @@ public void reschedule(JobEntity jobEntity, Date newDuedate) {
   }
 
   private void hintJobExecutorIfNeeded(JobEntity jobEntity, Date duedate) {
-    // Check if this timer fires before the next time the job executor will check for new timers to fire.
-    // This is highly unlikely because normally waitTimeInMillis is 5000 (5 seconds)
-    // and timers are usually set further in the future
-    JobExecutor jobExecutor = Context.getProcessEngineConfiguration().getJobExecutor();
-    int waitTimeInMillis = jobExecutor.getWaitTimeInMillis();
-    if (duedate.getTime() < (ClockUtil.getCurrentTime().getTime() + waitTimeInMillis)) {
+    if (duedate.getTime() < ClockUtil.getCurrentTime().getTime()) {
```

Comment:
Should probably be <=, or?

Reply:
Yes, correct! There are some failing tests and I'll include this with the fix.